test/Parse/ConditionalCompilation/basicParseErrors.swift

```diff
@@ -138,3 +138,9 @@ fn_k()
 func undefinedFunc() // ignored.
 #endif
 undefinedFunc() // expected-error {{use of unresolved identifier 'undefinedFunc'}}
+
+#if FOO
+#else if BAR
+// expected-error@-1 {{unexpected 'if' keyword following '#else' conditional compilation directive; did you mean '#elseif?'}}
+#else
+#endif
```

Comment:
Hmm. Could you also test that this is *not* treated as an error? I think the `Tok.isAtStartOfLine()` will handle this, but it doesn't hurt to check.

```swift
#if FOO
#else
  if true {}
#endif
```

Reply:
This works, buts definitely worth testing, thanks